vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -1406,9 +1425,14 @@ def fused_moe(
 
     if use_grouped_topk:
         assert num_expert_group is not None and topk_group is not None
-        topk_weights, topk_ids = grouped_topk(hidden_states, gating_output,
-                                              topk, renormalize,
-                                              num_expert_group, topk_group)
+        topk_weights, topk_ids = grouped_topk(
+            hidden_states,
+            gating_output,
+            topk,
+            renormalize,
+            num_expert_group,
+            topk_group,
+            share_fusion=envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS)
```

Comment:
Instead of accessing the global `envs.VLLM_SHARED_EXPERT_FUSION_REPLICAS` in a bunch of different places, could you instead only access it in the model definition and then pass it down from there?

Otherwise a very confusing situation could arise if these get decoupled, which would happen if somebody tries to set the share_fusion value in deepseek_v2 to anything else.